frontend/src/components/common/Button.vue

```diff
@@ -1,16 +1,24 @@
-<script setup>
+<script setup lang="ts">
 import Button from 'primevue/button';
+import type { PropType } from 'vue';
+import { IconPosition } from '../../enum/IconEnum';
 const props = defineProps({
     label: {
         type: String,
         default: 'Click',
     },
+    iconPos: {
+        type: String as PropType<IconPosition>,
+        default: IconPosition.right,
+    },
 });
 </script>
 <template>
-    <Button class="nr-button"
-        ><span>{{ props.label }}</span> <slot class="icon"></slot
-    ></Button>
+    <Button class="nr-button">
+        <slot class="icon" v-if="iconPos === IconPosition.left"></slot>
+        <span>{{ props.label }}</span>
+        <slot class="icon" v-if="iconPos === IconPosition.right"></slot>
```

Comment:
not so sure why we need these slot element?

Reply:
the slot is the icon, if the button receive left as props, the first slot (icon on will be before the text)  will be the one visible, else the second slot (icon will be after the text) will be the one visible.

Reply:
yes, got it, thanks for the explanation